### DISCUSSION TITLE
how to append to relation?

### DISCUSSION THREAD

**Discussion by hyusetiawan** [pocketbase/pocketbase#2802] [Q&A] [answered]
I can't seem to find a way to do this, is there a way to add to the many relations?
for ex: post -many -> comments
how do I add one more comment to the post? Right now the only way I can think of is to do the following:
1. query post with the array of comment ids
2. create the new comment
3. push the id of the newly created comment to the array of commentIds  in step 1
4. update post with it

is there an append command somewhere?

**Comment by ganigeorgiev** [comment, marked as answer]
The `relation` fields has `+`(append) and `-`(remove) modifiers support. You can use them like this:

```js
await pb.collection("example").update('RECORD_ID', {
    // append single id
    "yourRelField+": "id1",

    // or append multiple ids at once
    "yourRelField+": ["id1", "id2", "id3"],

    // remove single id
    "yourRelField-": "id1",

    // or remove multiple ids at once
    "yourRelField-": ["id1", "id2", "id3"],
})
```

_I'll consider documenting them better as currently the modifiers are only briefly mentioned in https://pocketbase.io/docs/collections/._

**Comment by alpcanm** [comment]
How can i do this in go lang side?

I tried this but in this case it behaves like a single field not list. It does not append , it changes the single value.

I want to append or remove.


```

record, err := config.App.Dao().FindRecordById("posts", postId)
	if err != nil {
		return err
	}

	record.Set("likes", userId)

	if err := config.App.Dao().SaveRecord(record); err != nil {
		return err
	}

```


> The `relation` fields has `+`(append) and `-`(remove) modifiers support. You can use them like this:
> 
> ```js
> await pb.collection("example").update('RECORD_ID', {
>     // append single id
>     "yourRelField+": "id1",
> 
>     // or append multiple ids at once
>     "yourRelField+": ["id1", "id2", "id3"],
> 
>     // remove single id
>     "yourRelField-": "id1",
> 
>     // or remove multiple ids at once
>     "yourRelField-": ["id1", "id2", "id3"],
> })
> ```
> 
> _I'll consider documenting them better as currently the modifiers are only briefly mentioned in https://pocketbase.io/docs/collections/._

**Reply by ganigeorgiev** [reply]
Appending or removing slice items is like for any Go slice value, aka. you have to retrieve the original "likes", modify the slice and then set it back to the record. Something like:

```go
likes := record.GetStringSlice("likes")

likes = append(likes, userId)

record.Set("likes", likes)
```

You can also make use of the Go slice helpers https://pkg.go.dev/slices.

**Reply by alpcanm** [reply]
Thank you
